kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/BaseConnectorIT.java

```diff
@@ -145,8 +147,18 @@ protected long waitForFilesInBucket(String bucketName, int numFiles) throws Inte
    * @return true if the number of files in the bucket match the expected number; false otherwise
    */
   protected Optional<Boolean> assertFileCountInBucket(String bucketName, int expectedNumFiles) {
+
+    int fileCount = 0;
+    ListObjectsV2Request request = new ListObjectsV2Request().withBucketName(bucketName);
+    ListObjectsV2Result result;
     try {
-      int fileCount = S3Client.listObjectsV2(bucketName).getKeyCount();
+      do {
+        result = S3Client.listObjectsV2(request);
+        fileCount += result.getKeyCount();
+        String token = result.getNextContinuationToken();
+        // To get the next batch of files.
+        request.setContinuationToken(token);
+      } while(result.isTruncated());
```

Comment:
Why can't we use the count below?
```
int fileCount = S3Client.listObjectsV2(bucketName).getKeyCount();
```
Do some of the expected files arrive with a delay?

Reply:
```       
 result = S3Client.listObjectsV2(request);
 fileCount += result.getKeyCount();
```
I have used the result object to store the result from the `S3Client.listObjectsV2(request)` so that I would be able extract the continuation token which is to be sent along with the request in order to get the next set of records.
The `listObjectsV2()` returns a maximum of 1000 files in a single iteration and thus we needed to use an additional `result` variable in order to get the next set of files(if they exist).

Reply:
Thanks for the clarification @SanchayGupta1197!